server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java

```diff
@@ -251,6 +251,10 @@ public IndexScopedSettings copy(Settings settings, IndexMetadata metadata) {
         return new IndexScopedSettings(settings, this, metadata);
     }
 
+    public boolean registerSetting(Setting<?> setting) {
```

Comment:
Not needed, method already public (https://github.com/opensearch-project/OpenSearch/pull/5495/files#diff-edf64b791aed5932da1f53fa264e42e98bd473f585ceed12892b66cda79c3fd7R153)